sequence_task_1/student/include/impl.h

```diff
@@ -20,31 +21,72 @@ class StaticPlaylist
 {
 public:
     /** @todo Member traits */
+    using value_type = typename Container::value_type;
+    using reference = typename Container::reference;
+    using const_reference = typename Container::const_reference;
+    using iterator = typename Container::iterator;
+    using const_iterator = typename Container::const_iterator;
+    using difference_type = typename Container::difference_type;
+    using size_type = typename Container::size_type;
 
     /** @todo Iterators */
+    const_iterator begin() {
+        return m_tracklist.begin();
+    }
+    const_iterator end() {
+        return m_tracklist.end();
+    }
+    const_iterator cbegin() {
+        return m_tracklist.cbegin();
+    }
+    const_iterator cend() {
+        return m_tracklist.cend();
+    }
 
     /** @todo Constructor from any reversible sequence container */
+    template<class AnyContainer>
+    StaticPlaylist(const AnyContainer& any_container)
+        : m_tracklist(any_container.begin(), any_container.end()) {}
 
     /** @todo Assignment from any reversible sequence container */
+    template<class AnyContainer>
+    StaticPlaylist& operator=(const AnyContainer& any_container) {
+        m_tracklist.assign(any_container.begin(), any_container.end());
+        return *this;
+    }
 
     /** @todo Add track from initializer */
     template<class... Args>
-    const Song& play(Args&&... songData);
+    const Song& play(Args&&... songData) {
+        return *m_tracklist.insert(m_tracklist.begin(), Song {std::forward<Args>(songData)...});
```

Comment:
You don't need to construct a temporary object, if you already have the initialization data, that can be used for in-place construction.